### ISSUE TITLE
Add more resampling functionality for Analog and IrregularlySampledSignals

### ISSUE THREAD

**Issue by JuliaSprenger** [NeuralEnsemble/python-neo#785]
Currently Signals can be resampled based on regular sampling intervals. Additionally it would make sense to be able to provide irregular sampling times and extract the corresponding values (via interpolation) from Analog as well as IrregularlySampledSignals. See also outdated `resample` method removed in #776.

**Comment by apdavison** [comment]
Probably more useful than _providing_ irregular sample times would be to have Neo calculate the sample times based on the time derivative of the signal (fewer samples when the signal is changing slowly...)

**Comment by JuliaSprenger** [comment]
Yes, that's also a good idea. However, this internally requires a method for sampling at specific times, which I think we can then also make accessible to the user, just in case this might be of interest.
